### ISSUE TITLE
How to detect multiple objects without specifying the number of objects to detect.

### ISSUE THREAD

**Issue by tomassosorio** [tomassosorio/DeepLearning#27]

**Comment by miguelriemoliveira** [comment]
Yes, this is a good question. From what you told me only a solution of specifying X outputs and then using a subset of X for the detections is possible. But what if we want more than X detections at some point ...

**Comment by tomassosorio** [comment]
Currently exists two solutions.
Segmentation and scan image.

Segmentation (R-CNN Paper, Fast R-CNN and Faster R-CNN papers):
First segment the image to see how many objects should we try to find.
And after create a object regression+classification to create a more accurate bounding box.
This takes a long time to predict, since first as to segment the image and after localize the object.

Scan (Paper YOLO, YOLO-v2, YOLO-v3):
You create a grid in the image and try to localize an object in each rectangle of the grid.
This approach appears to be slower since you have to scan all image but the results are much faster than the previously presented solution.

reference: https://www.coursera.org/learn/convolutional-neural-networks/home/welcome
